### PR TITLE
X22 device status

### PR DESCRIPTION
**Pull request by ps2**

### CODE REVIEW

File: RileyLink/DeviceDataManager.swift [hunk 1]
```diff
@@ -242,46 +370,45 @@ class DeviceDataManager {
 
     init() {
```
Comment (loudnate) [inline RileyLink/DeviceDataManager.swift:371]: Sorry to burden you with the haphazard expansion of Loop's DDM. I promise this will be refactored as things settle.

Comment (ps2) [inline RileyLink/DeviceDataManager.swift:371]: No worries; I'm just bringing RL up to speed with Loop, so when you do the refactoring, I can ~~steal~~ reuse your awesome ideas more easily. ;)

File: RileyLinkKit/PumpOpsSynchronous.swift [hunk 2]
```diff
@@ -435,19 +435,20 @@ class PumpOpsSynchronous {
...
 
 public struct PumpStatus {
-    let clock: NSDateComponents
-    let batteryVolts: Double
-    let batteryStatus: BatteryStatus
-    let suspended: Bool
-    let bolusing: Bool
-    let reservoir: Double
-    let model: PumpModel
+    public let clock: NSDateComponents
```
Comment (loudnate) [inline RileyLinkKit/PumpOpsSynchronous.swift:444]: ah sorry for not catching this sooner!

**Comment by loudnate** [comment]
LGTM ++

File: NightscoutUploadKit/NightscoutUploader.swift [hunk 3]
```diff
@@ -122,20 +122,6 @@ public class NightscoutUploader {
         uploadToNS(treatments, endpoint: defaultNightscoutTreatmentPath, completion: completionHandler)
     }
 
-    public func getPumpStatusFromMySentryPumpStatus(status: MySentryPumpStatusMessageBody) throws -> PumpStatus {
```
Comment (loudnate) [inline NightscoutUploadKit/NightscoutUploader.swift:125]: Why did you remove this, exactly? Is there an alternative method for uploading these values from MySentry records, or were you thinking you wanted all pumps to wake+poll (which I, personally, would not want to do)

Comment (ps2) [inline NightscoutUploadKit/NightscoutUploader.swift:125]: That was just the logic to convert MySentryPumpStatusMessageBody into PumpStatus.  I moved it into DDM, as it felt out of place in the uploader (really uses no state of the uploader).  https://github.com/ps2/rileylink_ios/blob/dev/RileyLink/DeviceDataManager.swift#L212-L226

Looking at that, I see I forgot to convert the throw.

Where would you suggest logic like this resides?  While I don't want to clutter up DDM too much, things that bridge frameworks (like this method bridges MinimedKit and NightscoutUploadKit models) seem to be ending up here (in DDM) at the high level, and that at least does have the advantage of keeping the frameworks focused.

Comment (ps2) [inline NightscoutUploadKit/NightscoutUploader.swift:125]: And yeah, I have no intention of not uploading from MySentry or forcing polling.

Comment (loudnate) [inline NightscoutUploadKit/NightscoutUploader.swift:125]: If you're OK with MinimedKit remaining a dependency on Nightscout upload it, then I'd make it a convenience initializer on PumpStatus. 

I didn't see it moved to DDM, sorry - I think that's a perfectly reasonable step in a path to allow NightscoutUloadKit to stand alone
